protostar-rust/src/lib.rs

```diff
@@ -99,6 +94,12 @@ pub fn run_test_runner(
         tests.len(),
     );
 
+    SnapboxCommand::new("scarb")
+        .arg("build")
+        .current_dir(std::env::current_dir().expect("failed to obtain current dir"))
+        .assert()
+        .success();
```

Comment:
I believe we could call it once at the beginning of the execution, right now we call it in the loop, once for each target.

Reply:
I also feel like we shouldn't use Snapbox here I created a task to change it #2082

Reply:
[updt](https://github.com/software-mansion/protostar/pull/2059/commits/2e3d9eb44a8158df46fa667e9b5f470643759b29)